platform/security/servlet/security-servlet-whoami/src/main/java/org/codice/ddf/security/servlet/whoami/WhoAmI.java

```diff
@@ -0,0 +1,167 @@
+
+import org.apache.shiro.subject.Subject;
+import org.joda.time.DateTime;
+import org.joda.time.Period;
+import org.joda.time.format.PeriodFormatter;
+import org.joda.time.format.PeriodFormatterBuilder;
+import org.opensaml.saml.saml2.core.AuthnContext;
+import org.opensaml.saml.saml2.core.AuthnContextClassRef;
+import org.opensaml.saml.saml2.core.AuthnStatement;
+
+import ddf.security.SubjectUtils;
+import ddf.security.assertion.SecurityAssertion;
+import ddf.security.principal.GuestPrincipal;
+
+public class WhoAmI {
+
+    private static final PeriodFormatter TIME_FORMATTER =
+            new PeriodFormatterBuilder()
+                    .appendDays()
+                    .appendSuffix(" day", " days")
+                    .appendSeparator(" ")
+                    .appendHours()
+                    .appendSuffix(" hour", " hours")
+                    .appendSeparator(" ")
+                    .appendMinutes()
+                    .appendSuffix(" minute", " minutes")
+                    .appendSeparator(" ")
+                    .appendSeconds()
+                    .appendSuffix(" second", " seconds")
+                    .toFormatter();
+
+    private final String name;
+
+    private final String displayName;
+
+    private final String email;
+
+    private final Date notOnOrAfter;
+
+    private final Date notBefore;
+
+    private final String expiresIn;
+
+    private final String issuer;
+
+    private final boolean isGuest;
+
+    private final List<String> authnContextClasses;
+
+    private final Map<String, SortedSet<String>> claims;
+
+    public WhoAmI(Subject subject) {
+        notNull(subject);
+
+        SecurityAssertion assertion = extractSecurityAssertion(subject);
+        notNull(assertion);
+
+        name = SubjectUtils.getName(subject);
+        displayName = SubjectUtils.getName(subject, null, true);
+        email = SubjectUtils.getEmailAddress(subject);
+        claims = SubjectUtils.getSubjectAttributes(subject);
+
+        issuer = assertion.getIssuer();
+        isGuest = assertion.getPrincipal() instanceof GuestPrincipal;
+
+        notOnOrAfter = assertion.getNotOnOrAfter();
+        notBefore = assertion.getNotBefore();
+        expiresIn = durationUntilNow(notOnOrAfter);
+
+        authnContextClasses = extractAuthnContextClasses(assertion.getAuthnStatements());
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getDisplayName() {
+        return displayName;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public String getIssuer() {
+        return issuer;
+    }
+
+    public Date getNotOnOrAfter() {
+        return new Date(notOnOrAfter.getTime());
+    }
+
+    public Date getNotBefore() {
+        return new Date(notBefore.getTime());
+    }
+
+    public String getExpiresIn() {
+        return expiresIn;
+    }
+
+    public List<String> getAuthnContextClasses() {
+        return authnContextClasses;
```

Comment:
It should not be a problem for this class but I went ahead and protected it in the constructor.